source/validations.ts

```diff
@@ -121,6 +136,37 @@ export class Validations {
 		})
 	}
 
+	/**
+	 * Ensures a given key is incremented only once per request
+	 * @param req
+	 * @param key
+	 */
+	singleCount(request: Request, store: Store, key: string) {
+		this.wrap(() => {
+			let stores = Validations.singleCountKeys.get(request)
+			if (!stores) {
+				stores = new Map()
+				Validations.singleCountKeys.set(request, stores)
+			}
+
+			const storeKey = store.localKeys ? store : store.constructor.name
```

Comment:
I thought the same, maybe we should change the PromisifiedStore to make it set the `constructor.name` property to `Promisified${originalStoreName}`?

Reply:
[This](https://stackoverflow.com/a/42660666) might be a case where `constructor.name` fails. We should follow the fix mentioned in the first comment below the answer, and advise other stores to do so as well.

Reply:
I was going to say that I think setting `constructor.name` on one instance would affect all other instances, but it looks like it's readonly:

```
> class Foo {}
undefined
> a = new Foo
Foo {}
> b = new Foo
Foo {}
> a.constructor.name
'Foo'
> b.constructor.name = 'bar'
'bar'
> a.constructor == b.constructor
true
> a.constructor.name == b.constructor.name
true
> b.constructor.name
'Foo'
```

As for the minification case, I think it would still work for us. They'd have to be minifiying server-side JS, and using two different stores, and the minifier would have to choose the same name for both stores, and the request would have to hit both stores for it to report a false positive.

I think I'm just going to update [the docs](https://github.com/express-rate-limit/express-rate-limit/wiki/Error-Codes#err_erl_double_count) to explain that it could be a false-positive in rare circumstances, and call this check good enough.

Reply:
Sounds good!